drivers/staging/vc04_services/interface/vchiq_arm/vchiq_core.c

```diff
@@ -587,8 +587,11 @@ reserve_space(VCHIQ_STATE_T *state, int space, int is_blocking)
 				return NULL; /* No space available */
 		}
 
-		BUG_ON(tx_pos ==
-			(state->slot_queue_available * VCHIQ_SLOT_SIZE));
+		if (tx_pos >= (state->slot_queue_available * VCHIQ_SLOT_SIZE)) {
```

Comment:
Is this change safe? Eventually these numbers will wrap, at which point the tail may be greater than the head. It is safer to check the difference between the two:
```
if (tx_pos - state->slot_queue_available * VCHIQ_SLOT_SIZE >= 0) {
```
with or without extra parentheses according to taste.